docs/demos/internal-developers/index.rst

```diff
@@ -0,0 +1,13 @@
+Development-targeted notebooks
```

Comment:
Currently this is in the middle of our demos contents; I think it should be right at the bottom so that it's clearer that it's not a normal demo

Reply:
I think putting at the end may require doing something slightly silly like naming the folder(s) `zzz-internal-developers`?